module/VuFind/src/VuFind/AjaxHandler/GetSearchResults.php

```diff
@@ -0,0 +1,455 @@
+    {
+        if ($params->fromQuery('searchType', '') === 'versions') {
+            // Customize the URL helper to make sure it builds proper versions URLs
+            $results->getUrlQuery()
+                ->setDefaultParameter('id', $params->fromQuery('id'))
+                ->setDefaultParameter('keys', $params->fromQuery('keys'))
+                ->setSuppressQuery(true);
+        }
+
+        $result = [];
+        foreach ($this->elements as $selector => $element) {
+            $content = call_user_func([$this, $element['method']], $params, $results);
+            if (null !== $content) {
+                $result[$selector] = [
+                    'content' => $content,
+                    'target' => $element['target'],
+                    'attrs' => $element['attrs'] ?? [],
+                ];
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Render search results
+     *
+     * @param Params  $params  Request params
+     * @param Results $results Search results
+     *
+     * @return ?string
+     */
+    protected function renderResults(Params $params, Results $results): ?string
+    {
+        [$baseAction] = explode('-', $results->getOptions()->getSearchAction());
+        $templatePath = "$baseAction/results-list.phtml";
+        if ('search' !== $baseAction && !$this->renderer->resolver($templatePath)) {
+            $templatePath = 'search/results-list.phtml';
+        }
+        $params = $results->getParams();
+        $options = $results->getOptions();
+        $cart = $this->renderer->plugin('cart');
+        $showBulkOptions = $options->supportsCart()
+            && ($this->config['Site']['showBulkOptions'] ?? false);
+        // Checkboxes if appropriate:
+        $showCartControls = $options->supportsCart()
+            && $cart()->isActive()
+            && ($showBulkOptions || !$cart()->isActiveInSearch());
+        // Enable bulk options if appropriate:
+        $showCheckboxes = $showCartControls || $showBulkOptions;
+
+        return $this->renderer->render(
+            $templatePath,
+            compact(
+                'results',
+                'params',
+                'showBulkOptions',
+                'showCartControls',
+                'showCheckboxes',
+            )
+        );
+    }
+
+    /**
+     * Render pagination
+     *
+     * @param Params  $params   Request params
+     * @param Results $results  Search results
+     * @param string  $template Paginator template
+     * @param string  $ulClass  Additional class for the pagination container
+     *
+     * @return ?string
+     */
+    protected function renderPagination(
+        Params $params,
+        Results $results,
+        string $template = 'search/pagination.phtml',
+        string $ulClass = ''
+    ): ?string {
+        $backend = $params->fromQuery('source', DEFAULT_SEARCH_BACKEND);
+        $paginationOptions = 'EDS' === $backend
```

Comment:
Can we perhaps move this logic into the Options class so we don't need an EDS-specific special case here?

Reply:
Options already has the `supportsFirstLastNavigation()` method, so I just switched to that. :)